docs/source/ops/index.rst

```diff
@@ -0,0 +1,86 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. ops-ref_index:
+
+DAML Participant pruning
+========================
+
+.. HINT::
+   DAML Participant pruning is currently an :doc:`Early Access Feature in Labs status </support/status-definitions>`
+
+The DAML Ledger API exposes an append-only ledger model; on the other hand, DAML Participants must be able to operate continuously for an indefinite amount of time on a limited amount of hot storage.
+
+In addition, privacy demands [1]_ may require removing Personally Identifiable Information (PII) upon request.
+
+To satisfy these requirements, the :ref:`Pruning Service <com.daml.ledger.api.v1.admin.ParticipantPruningService>` Ledger API endpoint [2]_ allows DAML Participants to support pruning of DAML contracts and transactions that were respectively archived and submitted before or at a given ledger offset.
+
+Please refer to the specific DAML Driver information for details about its pruning support.
+
+.. [1] For example, as enabled by provisions about the "right to be forgotten" of legislation such as
+       `EU's GDPR <https://gdpr-info.eu/>`_.
+.. [2] Invoking the Pruning Service requires administrative privileges.
+
+Impacts on DAML applications
+----------------------------
+
+When supported, pruning can be invoked by an operator with administrative privileges at any time on a healthy DAML participant; furthermore, it doesn't require stopping nor suspending normal operation.
+
+Still, DAML applications may be affected in the following ways:
+
+- Pruning is potentially a long-running operation and demanding one in terms of system resources; as such, it may significantly reduce DAML Ledger API throughput and increase latency while it is being performed. It is thus recommended to plan pruning invocations when low system utilization is expected.
+- Pruning may degrade the behavior of and/or crash the system if the pruning offset is too recent. In particular, the system might misbehave if command completions are pruned before the command trackers are able to process the completions.
```

Comment:
Good to call this out specifically as we had discussed with @oggy- . I am not sure if "crashing the system" best captures the worst-case scenario. Isn't the worst that can happen that completions would appear to block indefinitely?

Also is there an impact on command deduplication?

Perhaps this is also a good place to recommend that pruning should be run on "older" history than what is configured for command deduplication and command tracker retention?

Reply:
> I am not sure if "crashing the system" best captures the worst-case scenario. Isn't the worst that can happen that completions would appear to block indefinitely?

I rephrased to capture this a a slightly more general way.

Reply:
> Also is there an impact on command deduplication? Perhaps this is also a good place to recommend that pruning should be run on "older" history than what is configured for command deduplication and command tracker retention?

Seems sensible to me. @gerolf-da?

Reply:
yeah, makes sense. although this value is user supplied, so the participant operator likely should state their pruning window and applications need to adopt their timeouts correspondingly.

Reply:
Done.